### ISSUE TITLE
Bug in BugTripletModel calculate_loss method and evaluate method

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1936]
The BugTripletModel class has a bug in the calculate_loss method where the loss value is not always a float, and the evaluate method is not using torch.no_grad() block, which can cause performance issues and inaccuracies.
